WooCommerce/Classes/ViewRelated/Dashboard/Settings/POS/POSEligibilityChecker.swift

```diff
@@ -14,57 +16,88 @@ protocol POSEligibilityCheckerProtocol {
+        }
 
-    @Published private var isEligibleValue: Bool = false
+        return Publishers.CombineLatest(isOnboardingComplete, isWooCommerceVersionSupported)
+            .map { $0 && $1 }
+            .eraseToAnyPublisher()
+    }
 
     private let userInterfaceIdiom: UIUserInterfaceIdiom
     private let cardPresentPaymentsOnboarding: CardPresentPaymentsOnboardingUseCaseProtocol
     private let siteSettings: SelectedSiteSettings
     private let currencySettings: CurrencySettings
+    private let stores: StoresManager
     private let featureFlagService: FeatureFlagService
 
     init(userInterfaceIdiom: UIUserInterfaceIdiom = UIDevice.current.userInterfaceIdiom,
          cardPresentPaymentsOnboarding: CardPresentPaymentsOnboardingUseCaseProtocol = CardPresentPaymentsOnboardingUseCase(),
          siteSettings: SelectedSiteSettings = ServiceLocator.selectedSiteSettings,
          currencySettings: CurrencySettings = ServiceLocator.currencySettings,
+         stores: StoresManager = ServiceLocator.stores,
          featureFlagService: FeatureFlagService = ServiceLocator.featureFlagService) {
         self.userInterfaceIdiom = userInterfaceIdiom
         self.siteSettings = siteSettings
         self.currencySettings = currencySettings
         self.cardPresentPaymentsOnboarding = cardPresentPaymentsOnboarding
+        self.stores = stores
         self.featureFlagService = featureFlagService
-        observeOnboardingStateForEligibilityCheck()
     }
 }
 
 private extension POSEligibilityChecker {
-    /// Returns whether the selected store is eligible for POS.
-    func observeOnboardingStateForEligibilityCheck() {
-        // Conditions that are fixed for its lifetime.
-        let isTablet = userInterfaceIdiom == .pad
-        let isFeatureFlagEnabled = featureFlagService.isFeatureFlagEnabled(.displayPointOfSaleToggle)
-        guard isTablet && isFeatureFlagEnabled else {
-            isEligibleValue = false
-            return
-        }
-
-        cardPresentPaymentsOnboarding.statePublisher
+    var isOnboardingComplete: AnyPublisher<Bool, Never> {
+        return cardPresentPaymentsOnboarding.statePublisher
             .filter { [weak self] _ in
-                self?.isEligibleFromSiteChecks() ?? false
+                self?.isEligibleFromSiteChecks ?? false
             }
             .map { onboardingState in
                 // Woo Payments plugin enabled and user setup complete
                 onboardingState == .completed(plugin: .wcPayOnly) || onboardingState == .completed(plugin: .wcPayPreferred)
             }
-            .assign(to: &$isEligibleValue)
+            .eraseToAnyPublisher()
+    }
+
+    var isWooCommerceVersionSupported: AnyPublisher<Bool, Never> {
+        Future<Bool, Never> { [weak self] promise in
+            guard let self else {
+                promise(.success(false))
+                return
+            }
+
+            let siteID = stores.sessionManager.defaultStoreID ?? 0
```

Comment:
Perhaps we should add a guard and return early if we cannot get the `siteID`? Since we won't be able to continue anyway if this is 0. We can also add a `DDLogError` if we fall into the guard so the problem gets logged into the application logs.